packages/core/src/web/app/components/settings/AutoSave.spec.tsx

```diff
@@ -55,12 +55,40 @@ jest.mock('@core/app/widgets/PathInput', () => ({
   },
 }));
 
-jest.mock('@core/app/components/settings/SelectControl', () => ({ id, label, onChange, options }: any) => (
+const mockGetPreference = jest.fn();
+const mockSetPreference = jest.fn();
+
+const useSettingStore = create(() => ({
+  getPreference: mockGetPreference,
+  setPreference: mockSetPreference,
+}));
+
+jest.mock('@core/app/pages/Settings/useSettingStore', () => ({
+  useSettingStore,
+}));
+
+jest.mock('./components/SettingSelect', () => ({ id, label, onChange, options, url }: any) => (
   <div>
     mock-select-control id:{id}
     label:{label}
+    url:{url}
     options:{JSON.stringify(options)}
-    <input className="select-control" onChange={onChange} />
+    <input
+      className="select-control"
+      onChange={(e) =>
+        onChange(['false', 'true'].includes(e.target.value) ? e.target.value === 'true' : e.target.value)
+      }
+    />
+  </div>
+));
+
+jest.mock('./components/SettingFormItem', () => ({ children, id, label, options, url }: any) => (
+  <div>
+    mock-select-control id:{id}
+    label:{label}
+    url:{url}
+    options:{JSON.stringify(options)}
+    {children}
   </div>
 ));
```

Comment:
Is it possible to move `SettingSelect` & `SettingFormItem` mock to `__mock__`?
They are mocked several times with a little difference